server/konnectors/linkedin.js

```diff
@@ -0,0 +1,512 @@
+'use strict';
+
+const request = require('request');
+const https = require('https');
+const url = require('url');
+const cheerio = require('cheerio');
+const log = require('printit')({prefix: 'Linkedin', date: true});
+
+const localization = require('../lib/localization_manager');
+const Contact = require('../models/contact');
+const Tag = require('../models/tag');
+const ContactHelper = require('../lib/contact_helper');
+const CompareContacts = require('../lib/compare_contacts');
+const linkedin = require('../lib/linkedin_helper');
+
+const ACCOUNT_TYPE = 'com.linkedin';
+
+
+module.exports = {
+  name: 'Linkedin',
+  slug: 'linkedin',
+  description: 'konnector description linkedin',
+
+  fields: {
+    login: 'text',
+    password: 'password'
+  },
+
+  models: {
+    contact: Contact
+  },
+
+  fetch: (requiredFields, callback) => {
+    log.info('Import started');
+
+    Promise.resolve(requiredFields)
+    .then(retrieveTokens)
+    .then(logIn)
+    .then(retrieveContactList)
+    .then(prepareCozyContacts)
+    .then(getOrCreateTag)
+    .then(retrieveAndSaveContacts)
+    .then(createNotificationContent)
+    .then((notifContent) => {
+      callback(null, notifContent);
+    }, (reason) => {
+      callback(reason);
+    });
+  }
+};
+
+
+/**
+* Load landing page to retrieve the csrf token needed in login request.
+* More info on csrf -> https://en.wikipedia.org/wiki/Cross-site_request_forgery
+*
+* The html parsing is done with cheerio, a "jquery like" that can be run on
+* the server side.
+*/
+function retrieveTokens(requiredFields) {
+  return new Promise((resolve, reject) => {
```

Comment:
Sorry I meant the konnector developer, not the user.

When I read this code: https://github.com/pouchdb/pouchdb/blob/master/tests/mapreduce/test.persisted.js#L57-L108

I see that it's possible to chain many functions without calling the Promise constructor. So do you think it's possible to remove the `new Promise` in your function?